### PR TITLE
Basic elapsed time monitoring, estimated time remaining, and progress.

### PR DESCRIPTION
**Pull request by bluecamel**

### CODE REVIEW

File: stitcher/include/airmap/monitor/operation.h [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        value = value >= 0 ? value - 1 : count;
+        return {static_cast<Operation::Enum>(value)};
+    }
+
+    const Operation next() const
+    {
+        return {static_cast<Operation::Enum>((toInt() + 1) % count)};
+    }
+
+    const std::string str() const
```
Comment (remster) [inline stitcher/include/airmap/monitor/operation.h:93]: The class does read a bit like an overkill as enum itself behaves like most operators you have defined. It seems all you really need is this operator.

File: stitcher/include/airmap/monitor/timer.h [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+#include <chrono>
+#include <ostream>
+#include <tuple>
+#include <vector>
+
+namespace airmap {
+namespace stitcher {
+namespace monitor {
+
+class ElapsedTime {
```
Comment (remster) [inline stitcher/include/airmap/monitor/timer.h:12]: isn't this https://en.cppreference.com/w/cpp/chrono/duration the exact `std` equivalent of `ElapsedTime`?

File: stitcher/main.cpp [hunk 3]
```diff
@@ -29,6 +29,10 @@ int main(int argc, char *argv[])
             ("debug_path", boost::program_options::value<std::string>()->default_value("debug"),
                 "Path to the debug output.")
             ("disable_opencl", "If set, OpenCL will be disabled.")
+            ("elapsed_time", "Track elapsed times of the main stitch operations.  Always enabled if elapsed_time_log is.")
```
Comment (remster) [inline stitcher/main.cpp:32]: the purpose of this is unclear - why wouldn't the caller call `time stitcher ...` if they wish to know how long it took. Why won't the caller run its own timer and time incoming logs for progress. 
I am missing just basic percent progress here in the options.
And what's the difference between `estimate` and `estimate_log` as far as the caller is concerned?

File: stitcher/include/airmap/monitor/timer.h (hunk 2 above)
Comment (bluecamel) [reply stitcher/include/airmap/monitor/timer.h:12]: It's basically just a convenience wrapper around `std::duration`.

File: stitcher/main.cpp (hunk 3 above)
Comment (bluecamel) [reply stitcher/main.cpp:32]: The timer tracks elapsed times for individual stitch operations (e.g. undistortion, bundle adjustment, seam finder, etc.) and not just the entire stitch.

Enabling the estimator enables the calculations only.  For example, AB on Android deals with `LowLevelOpenCVStitcher` directly, so doesn't need logs to function.  However, AB on Linux runs the stitcher as a child process and reads logs to get estimate/progress updates.  One option is to enable estimation and the other adds logging.

**Pull request merged**